packages/xo-server/src/xapi-object-to-xo.js

```diff
@@ -54,12 +54,9 @@ function toTimestamp(date) {
     return timestamp
   }
 
-  const ms = parseDateTime(date)
-  if (!ms || ms.getTime() === 0) {
-    return null
-  }
+  const ms = parseDateTime(date)?.getTime()
 
-  return Math.round(ms.getTime() / 1000)
+  return ms === undefined || ms === 0 ? null : Math.round(ms / 1000)
```

Comment:
Why `undefined`?

Reply:
Because `ms` can be undefined if date doesn't respect good format.

Reply:
That's not what's written in the doc of [d3-time-format](https://devdocs.io/d3~5/d3-time-format#locale_parse).

Reply:
It appears I'm wrong and the optional chaining always returns `undefined` if there is a `null` or `undefined` in the path.

Reply:
Yep, you can keep your code, sorry about this :stuck_out_tongue: